brian2/devices/cpp_standalone/device.py

```diff
@@ -766,18 +767,23 @@ def copy_source_files(self, writer, directory):
                                   'random', 'randomkit', 'randomkit.h'),
                      os.path.join(directory, 'brianlib', 'randomkit', 'randomkit.h'))
 
+    def _insert_func_namespace(self, func, code_object, namespace):
+        impl = func.implementations[CPPStandaloneCodeObject]
+        func_namespace = impl.get_namespace(code_object.owner)
+        if func_namespace is not None:
+            namespace.update(func_namespace)
+        if impl.dependencies is not None:
+            for dep in impl.dependencies.itervalues():
+                self._insert_func_namespace(dep, code_object, namespace)
+
     def write_static_arrays(self, directory):
-        # # Find np arrays in the namespaces and convert them into static
-        # # arrays. Hopefully they are correctly used in the code: For example,
-        # # this works for the namespaces for functions with C++ (e.g. TimedArray
-        # # treats it as a C array) but does not work in places that are
-        # # implicitly vectorized (state updaters, resets, etc.). But arrays
-        # # shouldn't be used there anyway.
+        # Write Function namespaces as static arrays
         for code_object in self.code_objects.itervalues():
-            for name, value in code_object.variables.iteritems():
-                if isinstance(value, np.ndarray):
-                    self.static_arrays[name] = value
-                    
```

Comment:
Is the idea here that previously we looked for static arrays, but since the only way they can be accessed is by a function that we can just insert function namespaces instead?